applications/StructuralMechanicsApplication/custom_strategies/custom_schemes/explicit_central_differences_scheme.hpp

```diff
@@ -423,46 +444,84 @@ class ExplicitCentralDifferencesScheme
         #pragma omp parallel for schedule(guided,512)
         for (int i = 0; i < static_cast<int>(r_nodes.size()); ++i) {
             // Current step information "N+1" (before step update).
-            this->UpdateTranslationalDegreesOfFreedom(it_node_begin + i, disppos, dim);
+            this->UpdateTranslationalDegreesOfFreedom1(it_node_begin + i, disppos, dim);
+        } // for Node parallel
+
+        if (has_dof_for_rot_z){
```

Comment:
are u specializing to the case of having a rotation_z ?

why not updating all of the roatations at once?

Reply:
we are updating all of them but only if we have rotational dofs

Reply:
well "rot_z" as a name of variable is misleading...